pennylane/gradients/finite_difference.py

```diff
@@ -22,117 +22,155 @@
 
+    # sort columns in ascending order according to abs(shift)
+    coeffs_and_shifts = coeffs_and_shifts[:, np.argsort(np.abs(coeffs_and_shifts)[1])]
+    return coeffs_and_shifts
 
-def generate_shifted_tapes(tape, idx, shifts):
+
+def generate_shifted_tapes(tape, idx, shifts, multipliers=None):
     r"""Generate a list of tapes where the corresponding trainable parameter
     index has been shifted by the values given.
 
     Args:
         tape (.QuantumTape): input quantum tape
         idx (int): trainable parameter index to shift the parameter of
         shifts (Sequence[float or int]): sequence of shift values
+        multipliers (Sequence[float or int]): Sequence of multiplier values to
+            scale the paraameter by. If not provided, the parameter will
+            not be scaled.
 
     Returns:
         list[QuantumTape]: List of quantum tapes. Each tape has parameter
         ``idx`` shifted by consecutive values of ``shift``. The length
         of the returned list of tapes will match the length of ``shifts``.
     """
-    params = qml.math.stack(tape.get_parameters())
+    params = tape.get_parameters()
     tapes = []
 
-    for s in shifts:
+    for i, s in enumerate(shifts):
+        new_params = params.copy()
         shifted_tape = tape.copy(copy_operations=True)
 
-        shift = np.zeros(qml.math.shape(params), dtype=np.float64)
-        shift[idx] = s
-
-        shifted_params = params + qml.math.convert_like(shift, params)
-        shifted_tape.set_parameters(qml.math.unstack(shifted_params))
+        if multipliers is not None:
+            m = multipliers[i]
+            new_params[idx] = new_params[idx] * qml.math.convert_like(m, new_params[idx])
 
+        new_params[idx] = new_params[idx] + qml.math.convert_like(s, new_params[idx])
+        shifted_tape.set_parameters(new_params)
         tapes.append(shifted_tape)
 
     return tapes
 
 
-def finite_diff(tape, argnum=None, h=1e-7, order=1, n=1, form="forward"):
-    r"""Generate the parameter-shift tapes and postprocessing methods required
-    to compute the gradient of an gate parameter with respect to an
-    expectation value.
+def finite_diff(tape, argnum=None, h=1e-7, approx=1, n=1, strategy="forward", f0=None):
+    r"""Generate the finite-difference tapes and postprocessing methods required
+    to compute the gradient of a gate parameter with respect to its outputs.
 
     Args:
         tape (.QuantumTape): quantum tape to differentiate
         argnum (int or list[int] or None): Trainable parameter indices to differentiate
-            with respect to. If not provided, the derivative with respect to all
+            with respect to. If not provided, the derivatives with respect to all
             trainable indices are returned.
         h (float): finite difference method step size
-        order (int): The order of the finite difference method to use.
+        approx (int): The approximation order of the finite difference method to use.

```

Comment:
Check usage of "finite-difference" vs. "finite difference"

Reply:
```suggestion
        approx (int): The approximation order of the finite-difference method to use.
```